presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java

```diff
@@ -64,7 +64,12 @@ public KerberosAuthenticator(KerberosConfig config)
         System.setProperty("java.security.krb5.conf", config.getKerberosConfig().getAbsolutePath());
 
         try {
-            String hostname = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase(Locale.US);
+            String hostname = config.getHostName();
+
+            if (hostname == null || hostname.isEmpty()) {
+                hostname = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase(Locale.US);
```

Comment:
```
String hostname = Optional.ofNullable(config.getHostname())
  .orElseGet(() -> InetAddress.getLocalHost().getCanonicalHostName())
  .toLowerCase(Locale.US);
```

Reply:
@findepi Actually this would look more like

```suggestion
            String hostname = Optional.ofNullable(config.getHostName())
                    .orElseGet(() -> {
                        try {
                            return InetAddress.getLocalHost().getCanonicalHostName();
                        } catch (UnknownHostException e) {
                            throw new RuntimeException(e);
                        }
                    })
                    .toLowerCase(Locale.US);
```

since `getLocalHost()` throws `UnknownHostException()`. Do you think this is adequate or should I create a helper method as suggested by @electrum in https://github.com/prestodb/presto/pull/12126#discussion_r244562551?

More to the point should I even be working on this since you are asking @rclaude to submit his PR here?

I won't lie, this is pissing me off a little bit because prestodb/presto#12126 implements the exact same functionality in the exact same way yet my PRs were submitted first in both repos.

Reply:
> I won't lie, this is pissing me off a little bit because [prestodb/presto#12126](https://github.com/prestodb/presto/pull/12126) implements the exact same functionality in the exact same way yet my PRs were submitted first in both repos.

I understand that and I sympathize with you. I know this isn't fair and thank you that you're saying this frankly.
Of course it would be better to always review PRs in FIFO order, but it's not always the case.
Perhaps when you created your PR (https://github.com/prestodb/presto/pull/12047) people who could review this were busy doing something else.
And when @rclaude created his PR (https://github.com/prestodb/presto/pull/12126) it just happened that a potential reviewer had some free cycles.

Now, do you think it makes sense to go through the review process and have the same discussions again?

Reply:
@findepi A part of this situation won't make sense whichever perspective you take. Another way to look at this is that it doesn't make that much more sense to wait for prestodb/presto#12126 to be adapted when the discussions regarding this PR could be resolved simply with "Adopt the recommendations from prestodb/presto#12126 and add some docs", considering the fact that I am available to get it in shape _right now_.